tests/pos/t284-pos.scala

```diff
@@ -1,5 +1,5 @@
 trait B[T] {
-    def f1(a: T): Unit { }
+    //def f1(a: T): Unit { }
```

Comment:
Yes, that's temporary until someone figures out a better solution :). Refinement of unit creates a <refinement> class that extends Unit, since Unit extends AnyVal, this <refinement> class will be a value class. Either ExtensionMethods should not see this class, or it should include logic to skip it, or refining final classes shouldn't be allowed, I don't know what the best solution is.